htlcswitch/link.go

```diff
@@ -750,9 +757,12 @@ func (l *channelLink) htlcManager() {
 	// re-synchronize state with the remote peer. settledHtlcs is a map of
 	// HTLC's that we re-settled as part of the channel state sync.
 	if l.cfg.SyncStates {
-		if err := l.syncChanStates(); err != nil {
+		err = l.syncChanStates()
+		if err != nil {
 			l.errorf("unable to synchronize channel states: %v", err)
-			l.fail(err.Error())
+			if err != ErrLinkShuttingDown {
```

Comment:
I'm still not sure if this is the best way to handle this, it was added as a work around where the chan closer removes the link from the switch before it is able to send the shutdown message. Not sure if this is correct in all cases, any ideas?

Reply:
currently only line 586

Reply:
Re-visiting it again, this seems fine to me as if the quit channel of the link was closed, then it's likely also the case that the peer is being disconnected.